clcache/__main__.py

```diff
@@ -495,7 +513,7 @@ def _normalizedCommandLine(cmdline):
         argsToStrip += ("MP",)
 
         return [arg for arg in cmdline
-                if not (arg[0] in "/-" and arg[1:].startswith(argsToStrip))]
+                if arg[0] in "/-" and not arg[1:].startswith(argsToStrip)]
```

Comment:
Hmm, did you mean to change the behaviour here? I suspect the `and` should become `or` here. [De Morgan](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) found that `not (a and b)` is equivalent to `not a or not b`.

That being said, I'm not sure the new version is any nicer than the old one, so maybe it's not worth touching this at all.

Reply:
Oh, I only now noticed that you _meant_ to fix a bug here. In that case, can you please add a test case which demonstrates the bug?

Reply:
`testBasedirNoDirect` would fail without this bugfix, but I can add a test that exercises _only_ this bugfix and none of the other changes, if you like.